libdrgn/dwarf_info.c

```diff
@@ -5942,9 +5942,20 @@ drgn_dwarf_template_value_parameter_thunk_fn(struct drgn_object *res,
 static struct drgn_error *
 parse_template_parameter(struct drgn_debug_info *dbinfo,
```

Comment:
Since this only parses a template parameter if the `dwarf_tag()` matches, can we give it a different name? Something like `maybe_parse_template_parameter()` unless you have a better idea.